backend/src/realtime/realtime-note/realtime-note.service.spec.ts

```diff
@@ -89,15 +106,78 @@ describe('RealtimeNoteService', () => {
     setIntervalSpy = jest.spyOn(global, 'setInterval');
     clearIntervalSpy = jest.spyOn(global, 'clearInterval');
 
+    clientWithReadWrite = new MockConnectionBuilder(realtimeNote)
+      .withAcceptingRealtimeUserStatus()
+      .withLoggedInUser(readWriteUsername)
+      .build();
+
+    clientWithRead = new MockConnectionBuilder(realtimeNote)
+      .withDecliningRealtimeUserStatus()
+      .withLoggedInUser(onlyReadUsername)
+      .build();
+
+    clientWithoutReadWrite = new MockConnectionBuilder(realtimeNote)
+      .withDecliningRealtimeUserStatus()
+      .withGuestUser(noAccessUsername)
+      .build();
+
+    clientWithoutReadWrite.getTransporter().sendReady();
+    clientWithReadWrite.getTransporter().sendReady();
+
     realtimeNoteService = new RealtimeNoteService(
       revisionsService,
       consoleLoggerService,
       realtimeNoteStore,
       schedulerRegistry,
       mockedAppConfig,
+      mockedPermissionService,
     );
   });
 
+  it('should not remove the connection with read and write access', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+
+    const loggedUserTransporter = clientWithReadWrite.getTransporter();
+
+    jest.spyOn(loggedUserTransporter, 'disconnect');
+
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(loggedUserTransporter.disconnect).toHaveBeenCalledTimes(0);
+  });
+
+  it('should close the connection for removed connection', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+    const guestUserTransporter = clientWithoutReadWrite.getTransporter();
+    jest.spyOn(guestUserTransporter, 'disconnect');
+
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(guestUserTransporter.disconnect).toHaveBeenCalledTimes(1);
+  });
+
+  it('should update accept edit flag to true', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(clientWithReadWrite.acceptEdits).toBeTruthy();
+  });
+
+  it('should update accept edit flag to false', async () => {
+    jest.spyOn(realtimeNoteStore, 'find').mockImplementation(() => {
+      return realtimeNote;
+    });
+    await realtimeNoteService.handleNotePermissionChanged(note);
+
+    expect(clientWithRead.acceptEdits).toBeFalsy();
+  });
```

Comment:
There is no update happening. These values have been true/false before already. Easy fix: invert the accept values before running the handle function

Reply:
done, thanks